src/containerapp/azext_containerapp/_up_utils.py

```diff
@@ -825,6 +1024,18 @@ def _validate_up_args(cmd, source, artifact, image, repo, registry_server):
                                   "characters when using --repo")
 
 
+def _validate_custom_location_connected_cluster_args(cmd, env, resource_group_name, location, custom_location_id, connected_cluster_id):
+    if custom_location_id or connected_cluster_id:
+        register_provider_if_needed(cmd, EXTENDED_LOCATION_RP)
+        register_provider_if_needed(cmd, KUBERNETES_CONFIGURATION_RP)
```

Comment:
should we register RP only when there is related operations to do?

Reply:
```
    if custom_location_id or connected_cluster_id:
        register_provider_if_needed(cmd, EXTENDED_LOCATION_RP)
        register_provider_if_needed(cmd, KUBERNETES_CONFIGURATION_RP)
```
In this part, the customer input argument `--custom-location` or `--connected-cluster-id`, it will require register provider before executing the request.